### PR TITLE
adding test case for IDENTITY-3943

### PR DESCRIPTION
**Pull request by madurangasiriwardena**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+    public void initTest() throws Exception {
+
+        super.initTest();
+        applicationAuthenticationXml = new File(Utils.getResidentCarbonHome() + File.separator
+                + "repository" + File.separator + "conf" + File.separator
+                + "identity" + File.separator + "application-authentication.xml");
+        File applicationAuthenticationXmlToCopy = new File(FrameworkPathUtil.getSystemResourceLocation() +
+                "artifacts" + File.separator + "IS" + File.separator + "saml" + File.separator
+                + "application-authentication-changed-acs.xml");
+
```
Comment (kasungayan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java]: You may use Paths.get() in Java7 here to construct the path to these files.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+        super.stopCarbonServer(PORT_OFFSET_1);
+        super.stopTomcat(TOMCAT_8490);
+
+        super.stopHttpClient();
+
+        File applicationAuthenticationXmlToCopy = new File(FrameworkPathUtil.getSystemResourceLocation() +
+                "artifacts" + File.separator + "IS" + File.separator + "saml" + File.separator +
+                "application-authentication-default.xml");
+
```
Comment (kasungayan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java]: You may use Paths.get() in Java7 here to construct the path to these files.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+    private String sendSAMLResponseToPrimaryIS(HttpClient client, Map<String, String> searchResults) throws Exception {
+
+        HttpPost request = new HttpPost(String.format(COMMON_AUTH_URL, DEFAULT_PORT + PORT_OFFSET_0));
+        request.setHeader("User-Agent", USER_AGENT);
+
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("SAMLResponse", searchResults.get("SAMLResponse")));
+        urlParameters.add(new BasicNameValuePair("RelayState", searchResults.get("RelayState")));
+        request.setEntity(new UrlEncodedFormEntity(urlParameters));
```
Comment (kasungayan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java]: Try to use String constants.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java (hunk 1 above)
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java]: Fixed with https://github.com/wso2/product-is/pull/402

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java (hunk 2 above)
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java]: Fixed with https://github.com/wso2/product-is/pull/402

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java (hunk 3 above)
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java]: Fixed with https://github.com/wso2/product-is/pull/402

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java [hunk 4]
```diff
@@ -38,4 +42,11 @@ public static  boolean nameExists(FlaggedName[] allNames, String inputName) {
         return exists;
     }
 
+    public static String getResidentCarbonHome() {
+        if(StringUtils.isEmpty(RESIDENT_CARBON_HOME)){
```
Comment (cnapagoda) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java]: Formatting issue

Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java]: Fixed with https://github.com/wso2/product-is/pull/402

**Pull request merged**